modules/tinymce/src/themes/silver/main/ts/ui/core/ChoiceControls.ts

```diff
@@ -37,14 +37,23 @@ const registerController = <T>(editor: Editor, spec: ControlSpec<T>) => {
     const options = spec.getOptions(editor);
     const initial = spec.getCurrent(editor).map(spec.hash);
 
+    const current = Singleton.destroyable();
+
     return Arr.map(
       options,
-      (value, i) => ({
+      (value) => ({
         type: 'togglemenuitem',
         text: spec.display(value),
         onSetup: (api) => {
-          api.setActive(Optionals.is(initial, spec.hash(value)));
-          return spec.watcher(editor, value, api.setActive);
+          const setActive = (active: boolean) => {
+            api.setActive(active);
+            if (active) {
+              current.set({ destroy: () => api.setActive(false) });
+            }
+          };
+
+          setActive(Optionals.is(initial, spec.hash(value)));
+          return spec.watcher(editor, value, setActive);
```

Comment:
Does the singleton need to be destroyed here when the button is destroyed, otherwise this will potentially keep the API reference in memory indefinitely?

Reply:
The singleton should get garbage collected the same way all of the other state here does — the only things that hold a reference to it are the `setActive` functions we pass (through `spec.watcher`) into `FormatChanged`, and those call-backs gets removed automatically.